DirectProgramming/DPC++FPGA/ReferenceDesigns/gunzip/src/byte_stacker.hpp

```diff
@@ -0,0 +1,95 @@
+#ifndef __BYTE_STACKER_HPP__
+#define __BYTE_STACKER_HPP__
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/INTEL/ac_types/ac_int.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+
+#include "constexpr_math.hpp"
+
+using namespace sycl;
+
+//
+// Data streams in the 'InPipe' pipe and can have between 0 to
+// 'literals_per_cycle' valid elements per cycle. This function takes the input
+// and "stacks" it, such that the output is always 'literals_per_cycle' valid
+// elements (except for maybe the last write).
+//
+template <typename InPipe, typename OutPipe, unsigned literals_per_cycle>
+void ByteStacker() {
+  using OutPipeBundleT = decltype(OutPipe::read());
+  constexpr int cache_idx_bits = fpga_tools::Log2(literals_per_cycle * 2) + 1;

```

Comment:
This goes against what I was told by <person who does code formatting stuff>. Since it is 'computed' (albeit at compile time), it should not be CamelCase. Those seem to be reserved for "Magic Numbers", like `constexpr int kMyMagicNum = 7;` I can use the `k_` prefix though. Does that work?